src/ssl.jl

```diff
@@ -534,18 +534,22 @@ end
 function Base.unsafe_read(ssl::SSLStream, buf::Ptr{UInt8}, nbytes::UInt)
     nread = 0
     while nread < nbytes
-        (!isopen(ssl) || eof(ssl)) && throw(EOFError())
+        !isopen(ssl) && throw(EOFError())
         readbytes = ssl.readbytes
         @geterror ccall(
             (:SSL_read_ex, libssl),
             Cint,
-            (SSL, Ptr{Int8}, Csize_t, Ptr{Csize_t}),
+            (SSL, Ptr{UInt8}, Csize_t, Ptr{Csize_t}),
             ssl.ssl,
             buf + nread,
             nbytes - nread,
             readbytes
         )
-        nread += Int(readbytes[])
+        if ret == 1 || ret == SSL_ERROR_NONE
```

Comment:
Took me a while to follow this bit

so the SSL `ccall` returns `1` if successful, but if not successful, we don't return `0` we instead replace it with a `SSLErrorCode` enum?

I wonder if it'd be cleaner to always return a enum value? Then we could check just `ret == SSL_ERROR_NONE`, right?

We could still save ourselves a `SLL_get_error` call, by manually setting `SSL_ERROR_NONE` i.e. have `@geterror` check like:
```julia
if ret == 1
    ret = SSL_ERROR_NONE
else 
    err = get_error(...)
  ...
end
```